src/dune_rules/melange/melange_rules.mli

```diff
@@ -19,7 +19,15 @@ val setup_emit_js_rules :
   -> Melange_stanzas.Emit.t
   -> unit Memo.t
 
-val eval_runtime_deps :
-  expander:Expander.t -> Dep_conf.t list -> Path.Set.t Memo.t
+module Runtime_deps : sig
+  type path_specification =
+    | Allow_all
+    | Disallow_external of Lib_name.t
```

Comment:
If you want to be fancy, you could make this type:

```ocaml
type _ path_specification =
  | Allow_all : Path.Set.t path_specification
  | Disallow_external : Path.Build.Set.t path_specification
```

And then use it to type the return value

```ocaml
val eval : 'a path_specification -> 'a Memo.t
```